src/neural/cuda/layers.cc

```diff
@@ -755,12 +754,15 @@ void FusedWinogradConvSELayer<DataType>::LoadWeights(float* pfilter,
   const size_t weight_size = sizeof(float) * c_input_ * C * 3 * 3;
   const size_t blas_size = sizeof(float) * C;
 
+  // Store untransformed weights in scratch.
+  const DataType* weights = (DataType*)scratch + weight_size + blas_size;
```

Comment:
I'm guessing its unlikely, but shouldn't there be a logic somewhere that ensures that scratch is at least 2x weights size or something in order to do this safely?

Reply:
definitely - was just me being lazy. Updated.